### ISSUE TITLE
Write Tutorial

### ISSUE THREAD

**Issue by danielmewes** [rethinkdb/horizon-docs#6]
We currently have a quick start document https://github.com/rethinkdb/horizon-docs/blob/master/quickstart.md , but no real tutorial yet ( https://github.com/rethinkdb/horizon-docs/blob/master/tutorial/1-start.md )

**Comment by mglukhovsky** [comment]
@segphault is working on a basic tutorial that covers permissions and building a basic app.

@segphault: do you think the [Getting started] guide would overlap with the outline for the tutorial you're writing? Could you post the outline you're planning?

**Comment by segphault** [comment]
There are some differences in scope between the tutorial that I'm working on and the Getting Started guide. The tutorial will include:

- Building a social bookmark application with these features:
    - Users can add/edit/remove their own bookmarks
    - Users can share bookmarks into a public stream
    - Unauthenticated users can see public bookmarks
- It uses the following Horizon capabilities:
    - Authentication
        - hasAuthToken
        - currentUser
        - authEndpoint redirect
    - Permissions
        - Templates and validators that do the following:
            - Allow unauthenticated users to fetch public data
            - Ensure private data is only accessible to its owner
            - Allow users to update and delete their own data
    - Queries
        - findAll with order/limit
        - store
        - replace
        - remove
- Tutorial will additionally cover:
    - Writing the schema.toml and applying it from the command line
    - Using the auth methods and queries in a VueJS app

- The tutorial will not cover:
    - Step-by-step walkthrough of installing horizon and using hz init
    - Complete coverage of the relevant options in config.toml setup
    - Steps for generating and configuring the OAuth provider token

The TL;DR is that the tutorial is an application-focused introduction to using Horizon with permissions and authentication. It doesn't cover the basic setup, which I think is better left for documentation.
